src/command/backup/file.c

```diff
@@ -294,61 +294,84 @@ backupFile(
                     // Add size filter last to calculate repo size
                     ioFilterGroupAdd(ioReadFilterGroup(readIo), ioSizeNew());
 
-                    // Open the source and destination and copy the file
+                    // Open the source
                     if (ioReadOpen(readIo))
                     {
-                        // Setup the repo file for write. There is no need to write the file atomically (e.g. via a temp file on
-                        // Posix) because checksums are tested on resume after a failed backup. The path does not need to be synced
-                        // for each file because all paths are synced at the end of the backup. It needs to be created in the prior
-                        // context because it will live longer than a single loop when more than one file is being written.
-                        if (write == NULL)
+                        Buffer *const buffer = bufNew(ioBufferSize());
+
+                        do
                         {
-                            MEM_CONTEXT_PRIOR_BEGIN()
+                            ioRead(readIo, buffer);
+                            // Copy the file in non-bundling mode or if the file size is not zero
+                            if (bundleId == 0 || ioReadInterface(readIo)->totalBytesRead != 0)
                             {
-                                write = storageNewWriteP(
-                                    storageRepoWrite(), repoFile, .compressible = compressible, .noAtomic = true,
-                                    .noSyncPath = true);
-                                ioWriteOpen(storageWriteIo(write));
+                                // Setup the repo file for write. There is no need to write the file atomically (e.g.
+                                // via a temp file on Posix) because checksums are tested on resume after a failed
+                                // backup. The path does not need to be synced for each file because all paths are
+                                // synced at the end of the backup. It needs to be created in the prior context because
+                                // it will live longer than a single loop when more than one file is being written.
+                                if (write == NULL)
+                                {
+                                    MEM_CONTEXT_PRIOR_BEGIN()
+                                    {
+                                        write = storageNewWriteP(
+                                            storageRepoWrite(), repoFile, .compressible = compressible, .noAtomic = true,
+                                            .noSyncPath = true);
+                                        ioWriteOpen(storageWriteIo(write));
+                                    }
+                                    MEM_CONTEXT_PRIOR_END();
+                                }
+
+                                ioWrite(storageWriteIo(write), buffer);
```

Comment:
Seems like we could just use ioCopy() after the first buffer is written and get rid of the loop. This would make the code a bit more compact.

Reply:
Hmm, this may not work because of where the eof test is in ioCopy(). But that test could be moved to before the loop. Maybe getting too complicated.

Reply:
I may adjust this before commit, but it seems OK for now.